notebooks/machine_learning/raw/tut8.ipynb

```diff
@@ -0,0 +1,479 @@
+       "            min_samples_leaf=1, min_samples_split=2,\n",
+       "            min_weight_fraction_leaf=0.0, n_estimators=100, n_jobs=None,\n",
+       "            oob_score=False, random_state=7, verbose=0, warm_start=False)"
+      ]
+     },
+     "execution_count": 5,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Create the classifier and fit it to our training data\n",
+    "model = RandomForestClassifier(random_state=7, n_estimators=100)\n",
+    "model.fit(train_X, train_y)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The simplest metric for classification models is the **accuracy**, the number of correct predictions out of all predictions made. Scikit-learn provides `metrics.accuracy_score` to calculate this."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Accuracy: 0.864\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Predict classes given the validation features\n",
+    "pred_y = model.predict(val_X)\n",
+    "\n",
+    "# Calculate the accuracy as our performance metric\n",
+    "accuracy = metrics.accuracy_score(val_y, pred_y)\n",
+    "print(f\"Accuracy: {accuracy:.3f}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Confusion Matrix\n",
+    "\n",
+    "Our model did pretty well, correctly predicting around 86% of the price ranges in the validation data. It's often useful to look at where the model is failing with a **confusion matrix** which shows us how our model classified the inputs."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Confusion matrix:\n",
+      "[[130   6   0   0]\n",
+      " [  4  91  15   0]\n",
+      " [  0  21  98  12]\n",
+      " [  0   0  10 113]]\n",
+      "\n",
+      "Normalized confusion matrix:\n",
+      "[0.95588235 0.04411765 0.         0.        ]\n",
+      "[0.03636364 0.82727273 0.13636364 0.        ]\n",
+      "[0.         0.16030534 0.7480916  0.09160305]\n",
+      "[0.         0.         0.08130081 0.91869919]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Calculate the confusion matrix itself\n",
+    "confusion = metrics.confusion_matrix(val_y, pred_y)\n",
+    "print(f\"Confusion matrix:\\n{confusion}\")\n",
+    "\n",
+    "\n",
+    "# Normalizing by the true label counts to get rates\n",
+    "print(f\"\\nNormalized confusion matrix:\")\n",
+    "for row in confusion:\n",
+    "    print(row / row.sum())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<img src=\"https://i.imgur.com/idD0k8y.png\" alt=\"example confusion matrix\" width=400px>\n",
```

Comment:
I'd put a sentence like "In presentations, some people show this graphically like this:" above this graphic.

Following so closely on the output, I had to do a double take to see whether it was coming from the code... and a user may not know whether they are supposed to understand where this came from at this point.